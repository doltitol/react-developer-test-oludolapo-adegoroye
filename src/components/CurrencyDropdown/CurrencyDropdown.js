import React, { PureComponent } from 'react';
import './currencyoverlay.style.scss';
import { connect } from 'react-redux';
import { changeCurrency } from '../../redux/actions/cartActions';
import { useGQLQuery } from '../../graphql/useGQLQuery';

export class CurrencyDropdown extends PureComponent {
  state = {
    currencies: {},
  };

  handleCurrency = (currency) => {
    this.props.handleCurrency(currency);
  };
  getCurrencies = () => {
    useGQLQuery.currencies().then((result) =>
      this.setState({
        currencies: result,
      })
    );
  };
  getCurrencyDetails = () => {
    const currencies = this.state.currencies?.data.currencies;
    return { currencies };
  };
  componentDidMount() {
    this.getCurrencies();
  }
  render() {
    return (
      <>
        {this.state.currencies?.data && (
          <div className='currency-dropdown'>
            <ul className='currency-dropdown-list' data-testid='currency'>
              {this.getCurrencyDetails().currencies.map((currency) => (
                <li
                  key={currency.label}
                  className='currency-dropdown-list-item'
                  onClick={() => {
                    this.props.changeCurrency(currency.symbol);
                    this.props.closeCart();
                  }}
                >
                  <span>{currency.symbol}</span>
                  <span>{currency.label}</span>
                </li>
              ))}
            </ul>
          </div>
        )}
      </>
    );
  }
}
const mapStateToProps = (state) => {
  return {
    currency: state.cart.currency,
  };
};
const mapDispatchToProps = (dispatch) => {
  return {
    changeCurrency: (currency) => {
      dispatch(changeCurrency(currency));
    },
  };
};
export default connect(mapStateToProps, mapDispatchToProps)(CurrencyDropdown);
