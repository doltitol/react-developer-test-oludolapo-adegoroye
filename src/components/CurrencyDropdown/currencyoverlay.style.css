@import url("https://fonts.googleapis.com/css2?family=Raleway:wght@400;600;700&family=Roboto+Condensed:wght@700&family=Roboto:wght@400;500&family=Source+Sans+Pro&display=swap");
* {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  padding: 0;
  margin: 0;
  scroll-behavior: smooth;
}

body {
  background: #ffffff;
}

.slide-right {
  -webkit-animation: slide-right 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94) both;
  animation: slide-right 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94) both;
}

@-webkit-keyframes slide-right {
  0% {
    -webkit-transform: translateX(50px);
    transform: translateX(50px);
  }
  100% {
    -webkit-transform: translateX(0px);
    transform: translateX(0px);
  }
}

@keyframes slide-right {
  0% {
    -webkit-transform: translateX(50px);
    transform: translateX(50px);
  }
  100% {
    -webkit-transform: translateX(0px);
    transform: translateX(0px);
  }
}

.currency-dropdown {
  width: 100%;
  position: absolute;
  left: 0;
  right: 0;
  top: 80px;
  height: calc(100vh - 80px);
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
      -ms-flex-pack: end;
          justify-content: flex-end;
  padding: 0 70px;
}

.currency-dropdown-list {
  list-style: none;
  padding: 10px 0;
  width: 114px;
  max-height: 260px;
  background-color: #ffffff;
  -webkit-filter: drop-shadow(0px 4px 35px rgba(168, 172, 176, 0.19));
          filter: drop-shadow(0px 4px 35px rgba(168, 172, 176, 0.19));
}

.currency-dropdown-list-item {
  padding: 10px;
  text-align: center;
}

.currency-dropdown-list-item span {
  font-family: "Raleway", sans-serif;
  font-weight: 500;
  color: #1D1F22;
  font-size: 19px;
  line-height: 28.8px;
  padding: 0 2px;
  cursor: pointer;
  -webkit-transition: all 0.7s ease-in-out;
  transition: all 0.7s ease-in-out;
}

.currency-dropdown-list-item:hover {
  background-color: #EEEEEE;
}
/*# sourceMappingURL=currencyoverlay.style.css.map */