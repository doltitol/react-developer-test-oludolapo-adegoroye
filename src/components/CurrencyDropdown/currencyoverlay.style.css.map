{
    "version": 3,
    "mappings": "ACAA,OAAO,CAAC,wKAAI;AASZ,AAAA,CAAC,CAAC;EACA,UAAU,EAAE,UAAU;EACtB,OAAO,EAAE,CAAC;EACV,MAAM,EAAE,CAAC;EACT,eAAe,EAAE,MAAM;CACxB;;AACD,AAAA,IAAI,CAAA;EACJ,UAAU,ECZF,OAAO;CDad;;AACD,AAAA,CAAC,CAAA;EACD,eAAe,EAAE,IAAI;CACpB;;AACD,AAAA,YAAY,CAAC;EACX,iBAAiB,EAAE,WAAW,CAAC,IAAI,CAAC,oCAAoC,CAAC,IAAI;EAC7E,SAAS,EAAE,WAAW,CAAC,IAAI,CAAC,oCAAoC,CAAC,IAAI;CACtE;;AAED,kBAAkB,CAAlB,WAAkB;EAChB,EAAE;IACA,iBAAiB,EAAE,gBAAgB;IACnC,SAAS,EAAE,gBAAgB;;EAE7B,IAAI;IACF,iBAAiB,EAAE,eAAe;IAClC,SAAS,EAAE,eAAe;;;;AAI9B,UAAU,CAAV,WAAU;EACR,EAAE;IACA,iBAAiB,EAAE,gBAAgB;IACnC,SAAS,EAAE,gBAAgB;;EAE7B,IAAI;IACF,iBAAiB,EAAE,eAAe;IAClC,SAAS,EAAE,eAAe;;;;AAI9B,AAAA,WAAW,CAAC;EACV,iBAAiB,EAAE,UAAU,CAAC,IAAI,CAAC,oCAAoC,CAAC,IAAI;EAC5E,SAAS,EAAE,UAAU,CAAC,IAAI,CAAC,oCAAoC,CAAC,IAAI;CACrE;;AAED,kBAAkB,CAAlB,UAAkB;EAChB,EAAE;IACA,iBAAiB,EAAE,gBAAgB;IACnC,SAAS,EAAE,gBAAgB;;EAE7B,IAAI;IACF,iBAAiB,EAAE,eAAe;IAClC,SAAS,EAAE,eAAe;;;;AAI9B,UAAU,CAAV,UAAU;EACR,EAAE;IACA,iBAAiB,EAAE,gBAAgB;IACnC,SAAS,EAAE,gBAAgB;;EAE7B,IAAI;IACF,iBAAiB,EAAE,eAAe;IAClC,SAAS,EAAE,eAAe;;;;ADtE9B,AAAA,kBAAkB,CAAC;EACjB,KAAK,EAAE,IAAI;EACX,QAAQ,EAAE,QAAQ;EAClB,IAAI,EAAE,CAAC;EACP,KAAK,EAAE,CAAC;EACR,GAAG,EAAE,IAAI;EACT,MAAM,EAAE,kBAAkB;EACzB,OAAO,EAAE,IAAI;EACZ,eAAe,EAAE,QAAQ;EACzB,OAAO,EAAE,MAAM;EACjB,OAAO,EAAE,EAAE;CAuBZ;;AAtBE,AAAD,uBAAM,CAAC;EACL,UAAU,EAAE,IAAI;EAChB,OAAO,EAAE,MAAM;EACf,KAAK,EAAE,KAAK;EACZ,UAAU,EAAE,KAAK;EACjB,gBAAgB,EEbZ,OAAO;EFcX,MAAM,EAAE,mDAAmD;CAe5D;;AAbE,AAAD,4BAAM,CAAC;EACL,OAAO,EAAE,IAAI;EACb,UAAU,EAAE,MAAM;CAUnB;;AAZA,AAGC,4BAHI,CAGJ,IAAI,CAAC;EGfT,WAAW,EANC,SAAS,EAAE,UAAU;EAOjC,WAAW,EHe0B,GAAG;EGdxC,KAAK,EDRC,OAAO;ECSb,SAAS,EHac,IAAI;EGZ3B,WAAW,EHYkB,MAAM;EAC7B,OAAO,EAAE,KAAK;EACd,MAAM,EAAE,OAAO;EACf,UAAU,EAAE,oBAAoB;CACjC;;AARF,AASC,4BATI,AASH,MAAM,CAAC;EACN,gBAAgB,EEvBf,OAAO;CFwBT",
    "sources": [
        "currencyoverlay.style.scss",
        "../../styles/index.scss",
        "../../styles/_colors.scss",
        "../../styles/_fonts.scss",
        "../../styles/_colors.scss"
    ],
    "names": [],
    "file": "currencyoverlay.style.css"
}