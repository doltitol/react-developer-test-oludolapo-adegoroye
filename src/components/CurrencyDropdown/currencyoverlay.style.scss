@import '../../styles/index.scss';
.currency-dropdown {
  width: 100%;
  position: absolute;
  left: 0;
  right: 0;
  top: 80px;
  height: calc(100vh - 80px);
   display: flex;
    justify-content: flex-end;
    padding: 0 70px;
  &-list {
    list-style: none;
    padding: 10px 0;
    width: 114px;
    max-height: 260px;
    background-color: $white;
    filter: drop-shadow(0px 4px 35px rgba(168, 172, 176, 0.19));
   
    &-item {
      padding: 10px;
      text-align: center;
      span {
        @include raleway(19px, 28.8px, 500);
        padding: 0 2px;
        cursor: pointer;
        transition: all 0.7s ease-in-out;
      }
      &:hover {
        background-color: $bgGrey;
      }
    }
  }
}
