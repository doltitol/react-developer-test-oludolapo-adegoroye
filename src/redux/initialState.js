export const initialState = {
  cartItems: JSON.parse(localStorage.getItem('cartItems') || '[]'),
  cartQuantity: JSON.parse(localStorage.getItem('cartItems') || '[]').reduce(
    (acc, item) => acc + item.qty,
    0
  ),
  currency: '$',
};
