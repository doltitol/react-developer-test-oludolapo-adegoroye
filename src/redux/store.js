import { createStore, combineReducers } from 'redux';
import cartReducer from './reducers/cartReducers';

const rootReducer = combineReducers({
  cart: cartReducer,
});
export const store = createStore(rootReducer);
