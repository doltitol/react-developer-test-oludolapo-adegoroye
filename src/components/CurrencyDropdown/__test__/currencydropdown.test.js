import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { store } from '../../../redux/store';
import CurrencyDropdown from '../CurrencyDropdown';

describe('<CurrencyDropdown /> spec', () => {
  it('renders the empty component', () => {
    const container = render(
      <Provider store={store}>
        <CurrencyDropdown />
      </Provider>
    );
    expect(container.children).toMatchSnapshot();
  });
});
