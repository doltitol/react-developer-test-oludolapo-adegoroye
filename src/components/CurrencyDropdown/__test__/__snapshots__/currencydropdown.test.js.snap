// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrencyDropdown /> spec renders the empty component 1`] = `undefined`;
